src/service/url-replacements-impl.js

```diff
@@ -163,6 +164,14 @@ export class UrlReplacements {
       return removeFragment(info.sourceUrl);
     }));
 
+    this.setAsync_('SOURCE_URL', () => {
+      return getTrackImpressionPromise().then(
+        this.getDocInfoValue_.bind(this, info => {
```

Comment:
I don't have a good answer to the why. It is fine to read this directly inside of `UrlReplacements`, but you need to handle the shadow case.

`SOURCE_URL` and `QUERY_PARAM` should be the only affected, since the change cannot affect other parts of the URL, right?